main/inc/lib/sessionmanager.lib.php

```diff
@@ -5435,6 +5434,271 @@ public static function getCoursesListByCourseCoach($coachId)
     }
 
     /**
+     * Returns list of a few data from session (name, short description, start date, end date)
+     * And the next extra fields
+     * short_description, mode, human_text_duration, vacancies, brochure, target, schedule
+     * from Session category Id.
+     * @param int $categoryId
+     * @param string $target
+     * @return mixed
+     */
+    public static function getBriefSessionListAndExtraByCategory($categoryId, $target) {
+        // Init variables
+        $categoryId = (int) $categoryId;
+        $sessionList = array();
+        // Check if categoryId is valid
+        if ($categoryId > 0) {
+            $target = Database::escape_string($target);
+            $sTable = Database::get_main_table(TABLE_MAIN_SESSION);
+            $sfTable = Database::get_main_table(TABLE_MAIN_SESSION_FIELD);
+            $sfvTable = Database::get_main_table(TABLE_MAIN_SESSION_FIELD_VALUES);
+            // Join session field and session field values tables
+            $joinTable = $sfTable . ' sf INNER JOIN ' . $sfvTable . ' sfv ON sf.id = sfv.field_id';
+            $fieldsArray = array(
+                'short_description', 'mode', 'human_text_duration', 'vacancies', 'brochure', 'target', 'schedule'
+            );
+            // Get the session list from session category and target
+            $sessionList = Database::select(
+                'id, name, date_start, date_end',
+                $sTable,
+                array(
+                    'where' => array(
+                        "session_category_id = ? AND id IN (
+                            SELECT sfv.session_id FROM $joinTable WHERE
+                            sfv.session_id = session.id
+                            AND sf.field_variable = 'target'
+                            AND sfv.field_value = ?
+                        );" => array($categoryId, $target)
+                    )
+                )
+            );
+
+            // Get session fields
+            $extraField = new ExtraField('session');
+            $fieldList = $extraField->get_all(array(
+                'field_variable IN ( ?, ?, ?, ?, ?, ?, ? )' => $fieldsArray
+            ));
+            // Index session fields
+            foreach ($fieldList as $field) {
+                $fields[$field['id']] = $field['field_variable'];
```

Comment:
Make sure you always *define* an array before starting to use it. $fields was not defined before this foreach